src/io/flutter/project/FlutterProjectOpenProcessor.java

```diff
@@ -87,18 +94,27 @@ private void doPostOpenProcessing(@Nullable Project project) {
       return;
     }
 
-    // TODO(pq): handle adding Flutter module type.
-    if (!FlutterSdkUtil.hasFlutterModule(project)) {
-      return;
+    if (!FlutterModuleUtils.hasFlutterModule(project)) {
+      if (FlutterModuleUtils.usesFlutter(project)) {
+        final List<Module> modules = FlutterModuleUtils.findModulesWithFlutterContents(project);
+        if (modules.isEmpty()) {
+          LOG.warn(MessageFormat.format("No module found for {0}", project.getName()));
+          return;
+        } else if (modules.size() > 1){
```

Comment:
It's an odd case.  Essentially it means we've got 2 modules that have pubspecs with flutter deps (or the module type declared) and we're just picking the first one.  I don't expect this to happen often (or ever) but in case it does it may be nice to log something for the record.